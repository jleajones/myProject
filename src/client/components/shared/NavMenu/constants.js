// export default ['how it works', 'ship with us', 'drive with us', 'login'];

export default [
  {
    display: 'how it works'
  },
  {
    display: 'ship with us',
    isModal: true
  },
  {
    display: 'drive with us'
  },
  {
    display: 'login',
    isModal: true
  }
];
