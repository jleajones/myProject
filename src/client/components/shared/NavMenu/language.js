export default {
  driveWithUs: () => 'drive with us',
  howItWorks: () => 'how it works',
  login: () => 'login',
  shipWithUs: () => 'ship with us'
};
