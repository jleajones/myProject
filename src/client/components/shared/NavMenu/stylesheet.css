.navContainer ul {
  margin: 0;
}
