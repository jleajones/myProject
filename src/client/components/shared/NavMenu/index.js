import React from 'react';
import styled from 'styled-components';
import { camelCase } from '@clientLib/utils';

import Link from '@components/Link';
import PATHS from '@constants/paths';

import { items } from './constants';
import lang from './language';
import styles from './stylesheet.css';


const NavMenu = () => (
  <nav className={styles.navContainer}>
    <ul>
      {items.map(item => (
        <NavItem key={item}>
          <Link
            to={PATHS[camelCase(item)].path}
            category={'Header'}
            label={item}
          >
            {lang[camelCase(item)]()}
          </Link>
        </NavItem>
      ))}
    </ul>
  </nav>
);

const NavItem = styled.li`
  color: ${props => props.theme.colors.primary}
`;

export default NavMenu;
