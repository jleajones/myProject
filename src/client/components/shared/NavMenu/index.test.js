import React from 'react';
import { shallow } from 'enzyme';

import NavMenu from '.';

describe('<NavMenu />', () => {
  test('Renders component without crashing', () => {
    const handleCloseDialog = jest.fn();
    const handleOpenDialog = jest.fn();
    const open = false;
    const component = shallow(
      <NavMenu
        handleCloseDialog={handleCloseDialog}
        handleOpenDialog={handleOpenDialog}
        open={open}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
