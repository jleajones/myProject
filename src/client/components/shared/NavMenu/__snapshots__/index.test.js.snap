// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavMenu /> Renders component without crashing 1`] = `
<nav
  className="makeStyles-root-1"
>
  <Connect(withRouter(Link))
    category="header"
    className="makeStyles-link-2"
    key="how it works"
    label="how-it-works"
    onClick={[Function]}
    to="/how-it-works"
  >
    how it works
  </Connect(withRouter(Link))>
  <Connect(withRouter(Link))
    category="header"
    className="makeStyles-link-2"
    key="ship with us"
    label="ship-with-us"
    onClick={[Function]}
    to={Object {}}
  >
    ship with us
  </Connect(withRouter(Link))>
  <Connect(withRouter(Link))
    category="header"
    className="makeStyles-link-2"
    key="drive with us"
    label="drive-with-us"
    onClick={[Function]}
    to="/drive-with-us"
  >
    drive with us
  </Connect(withRouter(Link))>
  <Connect(withRouter(Link))
    category="header"
    className="makeStyles-link-2"
    key="login"
    label="login"
    onClick={[Function]}
    to={Object {}}
  >
    login
  </Connect(withRouter(Link))>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-button-3"
    size="small"
    variant="outlined"
  >
    Call Us Today
  </WithStyles(ForwardRef(Button))>
  <LoginDialog
    handleClose={[Function]}
    open={false}
  />
  <CarrierSignUp
    handleClose={[Function]}
    open={false}
  />
</nav>
`;
